internal/encryption/keyprovider/openbao/client.go

```diff
@@ -0,0 +1,96 @@
+package openbao
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	openbao "github.com/openbao/openbao/api"
+)
+
+type client interface {
+	WriteBytesWithContext(ctx context.Context, path string, data []byte) (*openbao.Secret, error)
+}
+
+// service implements missing utility functions from openbao/api such as routing and serialization.
+type service struct {
+	c client
+}
+
+type dataKey struct {
+	Plaintext  []byte
+	Ciphertext []byte
+}
+
+func (s service) generateDataKey(ctx context.Context, keyName string, bitSize int) (dataKey, error) {
+	path := fmt.Sprintf("/transit/datakey/plaintext/%s", keyName)
```

Comment:
This looks like a URL. Do we need to urlencode the key name?

Reply:
Yes, it's a URL, but we don't need to additionally escape it. `openbao/api` uses `net.URL` under the hood to escape the whole path: https://github.com/openbao/openbao/blob/main/api/client.go#L1193